types/urllib/index.d.ts

```diff
@@ -0,0 +1,253 @@
+
+/**
+ * yield urllib.requestThunk(url, args)
+ * @param url The URL to request, either a String or a Object that return by url.parse.
+ * @param options Optional, @see RequestOptions.
+ */
+export function requestThunk(url: string | url.URL, options: RequestOptions): (callback: (...args: any[]) => void) => void;
+
+/**
+ * alias to request.
+ * Handle all http request, both http and https support well.
+ *
+ * @example
+ * // GET http://httptest.cnodejs.net
+ * urllib.request('http://httptest.cnodejs.net/test/get', function(err, data, res) {});
+ * // POST http://httptest.cnodejs.net
+ * var args = { type: 'post', data: { foo: 'bar' } };
+ * urllib.request('http://httptest.cnodejs.net/test/post', args, function(err, data, res) {});
+ *
+ * @param url The URL to request, either a String or a Object that return by url.parse.
+ * @param options Optional, @see RequestOptions.
+ */
+export function curl(url: string | url.URL, options?: RequestOptions): Promise<any>;
+/**
+ * @param url The URL to request, either a String or a Object that return by url.parse.
+ * @param callback @see Callback
+ */
+export function curl(url: string | url.URL, callback: Callback): void;
+/**
+ * @param url The URL to request, either a String or a Object that return by url.parse.
+ * @param options Optional, @see RequestOptions.
+ * @param callback @see Callback
+ */
+export function curl(url: string | url.URL, options: RequestOptions, callback: Callback): void;
+/**
+ * The default request timeout(in milliseconds).
+ * @const
+ */
+export const TIMEOUT: number;
+/**
+ * The default request & response timeout(in milliseconds).
+ * @const
+ */
+export const TIMEOUTS: [number, number];
+
+/**
+ * Request user agent.
+ * @const
+ */
+export const USER_AGENT: string;
+
+/**
+ * Request http agent.
+ */
+export const agent: http.Agent;
+
+/**
+ * Request https agent.
+ */
+export const httpsAgent: https.Agent;
+
+export class HttpClient extends EventEmitter {
+  constructor(options?: RequestOptions);
+
+  request(url: string | url.URL, callback: Callback): void;
+  request(url: string | url.URL, options: RequestOptions, callback: Callback): void;
+
+  curl(url: string | url.URL, callback: Callback): void;
+  curl(url: string | url.URL, options: RequestOptions, callback: Callback): void;
+
+  requestThunk(url: string | url.URL, options?: RequestOptions): (callback: (...args: any[]) => void) => void;
+}
+
+/**
+ * request method only return a promise,
+ * compatible with async/await and generator in co.
+ */
+export class HttpClient2 extends EventEmitter {
+  constructor(options?: RequestOptions);
+
+  request(url: string | url.URL, options?: RequestOptions): Promise<any>;
+
+  curl(url: string | url.URL, options?: RequestOptions): Promise<any>;
+
+  requestThunk(url: string | url.URL, options?: RequestOptions): (callback: (...args: any[]) => void) => void;
+}
+
+/**
+ * Create a HttpClient incetance.
+ * @param options
+ * @return HttpClient incetance.
```

Comment:
This is also redundant -- the return type is `HttpClient`, so of course an instance of it is returned.